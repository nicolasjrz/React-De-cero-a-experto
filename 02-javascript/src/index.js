const getImagen = async () => {
  const api_key = "gIbgAxARtogmrtPUWSZQARUt07Q9g05k";
  const resp = await fetch(
    `https://api.giphy.com/v1/gifs/random?api_key=${api_key}`
  );
  const { data } = await resp.json();
  const { url } = data.images.original;
  imprimirImagen(url);
};

const imprimirImagen = (url) => {
  const img = document.createElement("img");
  img.src = url;
  document.body.append(img);
};

getImagen();
