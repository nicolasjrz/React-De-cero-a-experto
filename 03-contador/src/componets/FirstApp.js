import React from "react";

export const FirstApp = () => {
  return (
    <>
      <h1>Hellow word app</h1>
      <p>jeje holis</p>
    </>
  );
};
