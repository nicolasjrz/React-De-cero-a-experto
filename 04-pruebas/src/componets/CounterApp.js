import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setcounter] = useState(value);

  const handleSuma = () => {
    setcounter(counter + 1);
  };

  const handleResta = () => {
    if (counter > value) {
      setcounter(counter - 1);
    }
  };

  const handleReset = () => {
    setcounter(value);
  };
  return (
    <>
      <h1>counter app </h1>
      <h2>{counter}</h2>
      <button onClick={handleSuma}>+1</button>
      <button onClick={handleResta}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 1,
};

export default CounterApp;
