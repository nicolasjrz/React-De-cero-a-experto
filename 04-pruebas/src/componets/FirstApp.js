import React from "react";
import PropTypes from "prop-types";
const FirstApp = ({ title, subtitle, parrafo }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>

      <p>{subtitle}</p>
      <p>{parrafo}</p>
    </>
  );
};

export default FirstApp;

FirstApp.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
  //title: "Prueba de react",
  subtitle: "no hay subtitulo",
  parrafo: "no hay parrafo",
};
