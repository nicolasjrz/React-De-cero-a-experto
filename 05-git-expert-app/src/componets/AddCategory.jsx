import React, { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ onNewCategory }) => {
  const [value, setValue] = useState("");

  const handleInputChange = ({ target }) => {
    setValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (value.trim().length <= 1) return;
    //setCategories((categories) => [value, ...categories]);
    onNewCategory(value.trim());
    setValue("");
  };
  return (
    <form aria-label="form" onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        name="input"
        id="input"
        placeholder="search gift"
        onChange={handleInputChange}
        value={value}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
