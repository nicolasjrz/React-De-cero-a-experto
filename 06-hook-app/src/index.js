import React from "react";
import ReactDOM from "react-dom/client";
//import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
import "./index.css";
//import { HooksApp } from "./HooksApp";
//import { CounterWithCustomHook } from "./CounterWithCustomHook";
//import { SimpleForm } from "./useEfect/SimpleForm";
//import { FormWithCustomHook } from "./useEfect/FormWithCustomHook";
//import { FocusScreen } from "./04-useRef/FocusScreen";
//import { MemoHook } from "./06-memos/MemoHook";
//import { CallbackHook } from "./07-Callbacks/CallbackHook";
//import { Padre } from "./07-tarea-memo/Padre";
import { TodoApp } from "./09-useReducer/TodoApp";
const root = ReactDOM.createRoot(document.getElementById("root"));
root.render(
  //<React.StrictMode>
  <TodoApp />
  //</React.StrictMode>
);
