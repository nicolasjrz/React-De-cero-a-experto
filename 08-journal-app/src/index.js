import React from "react";
import ReactDOM from "react-dom/client";
import "./style.css";
import { JournalApp } from "./JournalApp";
import { BrowserRouter } from "react-router-dom";
import { store } from "./store/store";
import { Provider } from "react-redux";
const root = ReactDOM.createRoot(document.getElementById("root"));
root.render(
  <React.StrictMode>
    <Provider store={store}>
      <BrowserRouter>
        <JournalApp />
      </BrowserRouter>
    </Provider>
  </React.StrictMode>
);
