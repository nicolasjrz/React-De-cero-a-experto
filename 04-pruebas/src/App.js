import "./App.css";
//import FirstApp from "./componets/FirstApp";
import CounterApp from "./componets/CounterApp";
function App() {
  return (
    <div className="App">
      <CounterApp value={10} />
    </div>
  );
}

export default App;
