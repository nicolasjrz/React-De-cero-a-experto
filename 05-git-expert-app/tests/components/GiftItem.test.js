import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/componets";

describe("Test de gift item", () => {
  const url = "http://localhost/wwww";
  const title = "titulo";
  const alt = "titulo";
  test("debe haer match con el snapshot  ", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el alt indicado ", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    //expect(screen.getByRole("img").src).toBe(url);
    //console.log(screen.getByRole("img").src);
    //console.log("hgola");
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
