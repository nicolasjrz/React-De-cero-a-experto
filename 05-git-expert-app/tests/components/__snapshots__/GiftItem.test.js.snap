// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de gift item debe haer match con el snapshot   1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="http://localhost/wwww"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
